### PR TITLE
Updated the grammar rules to allow for optional trailing commas.

### PR DESCRIPTION
**Pull request by MahtabNorouzi**
- [ ] Tests added for any new code
- [ ] Documentation added for any new functionality
- [ ] Entries added to the respective `CHANGELOG.md` for any new functionality
- [ ] Feature table on [`README.md`](../README.md#roadmap) updated for any listed functionality

### CODE REVIEW

File: quint/src/generated/Quint.g4 [hunk 1]
```diff
@@ -92,7 +92,7 @@ exportMod : 'export' name '.' identOrStar
 // an instance may have a special parameter '*',
 // which means that the missing parameters are identity, e.g., x = x, y = y
 instanceMod :   // creating an instance and importing all names introduced in the instance
-                'import' moduleName '(' (name '=' expr (',' name '=' expr)*) ')' '.' '*'
+                'import' moduleName '(' (name '=' expr (',' name '=' expr)*) ','? ')' '.' '*'
                   ('from' fromSource)?
                 // creating an instance and importing all names with a prefix
             |   'import' moduleName '(' (name '=' expr (',' name '=' expr)*) ')' 'as' qualifiedName
```
Comment (bugarela) [inline quint/src/generated/Quint.g4:98]: We should also allow the comma on this form to be consistent

**Review by bugarela — COMMENTED** [review]
We should add a test for this, such as adding some usages of all new forms of trailing commas to `SuperSpec.qnt` - and then updating `SuperSpec.json` by running `npm run update-fixtures`

**Comment by MahtabNorouzi** [comment]
## Summary of Changes
- Updated the grammar rules to allow for optional trailing commas.
- Modified the `min` function in `superspec.map.json` to demonstrate this change while keeping the `add` function unchanged for reference.
- Added an import statement: `import Proto(N = N,) as Inst1`.

## Details
- The `min` function now accepts parameters in the following formats:
  - `(x: int, y: int,)`
  - `(x: int, y: int)`
- The import statement is included to utilize the `Proto` module with an optional trailing comma.

## Reason for Changes
- Allowing an optional trailing comma makes the syntax more flexible and consistent with common programming practices.

## Test Cases
- Added test cases for both scenarios in `superspec.map.json` to ensure correct parsing of functions with optional trailing commas.

**Review by bugarela — APPROVED** [review]
Looks great! Before merging it, can you add "Closes #1383" to the PR description, so the issue gets automatically closed when we merge this?

**Comment by bugarela** [comment]
Actually, I remembered something else: we should add an entry to CHANGELOG.md saying that we now allow trailing commas in operator definitions and constant initialization - which also makes me think we might have missed one case: operator call. We should be able to call `min(1, 2,)` just as well as we can define `min(x: int, y: int,)`

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,25 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Added
+
+- Trailing commas are now allowed in operator definitions and constant initialization (#1510)
+
+### Changed
+
+- Updated grammar rule to allow an optional trailing comma in parameter lists:
+  
+  - Function calls
```
Comment (bugarela) [inline CHANGELOG.md:28]:
```suggestion
  - Operator calls
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,11 +12,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `quint verify` has the option `--apalache-version` to pull a custom version (#1521)
 
 ### Changed
+
+- Updated grammar rule to allow an optional trailing comma in parameter lists:
```
Comment (bugarela) [inline CHANGELOG.md:16]: It's good to add a reference to an issue/PR for whoever needs more context and details.
```suggestion
- Updated grammar rule to allow an optional trailing comma in parameter lists (#1510):
```

**Review by bugarela — APPROVED** [review]
Looks great! I've just merged another PR that touches the grammar, so you'll need to re-generate the files once merging main on this - sorry about that! Otherwise, we're good to go :sparkles:

**Pull request merged**